file/sst_file_manager_impl.h

```diff
@@ -36,7 +36,11 @@ class SstFileManagerImpl : public SstFileManager {
   ~SstFileManagerImpl();
 
   // DB will call OnAddFile whenever a new sst file is added.
-  Status OnAddFile(const std::string& file_path, bool compaction = false);
+  // Parameter `file_size` is for optimization: if the caller knows the size of
+  // the file, they can provide it, otherwise OnAddFile() will query it from
+  // the filesystem.
+  Status OnAddFile(const std::string& file_path, bool compaction = false,
```

Comment:
I prefer to just overload OnAddFile() rather than add another default parameter.